service/frontend/workflow_handler.go

```diff
@@ -4662,3 +4687,72 @@ func getBatchOperationState(workflowState enumspb.WorkflowExecutionStatus) enums
 	}
 	return operationState
 }
+
+func (wh *WorkflowHandler) mapStartWorkflowExecutionRequestSearchAttributes(request *workflowservice.StartWorkflowExecutionRequest, namespaceName namespace.Name) (*workflowservice.StartWorkflowExecutionRequest, error) {
+	mappedSearchAttributes, err := searchattribute.SubstituteAliases(wh.saMapper, request.GetSearchAttributes(), namespaceName.String())
+	if err != nil {
+		return nil, err
+	}
+	if mappedSearchAttributes == nil {
+		return request, nil
+	}
+
+	// Shallow copy request and replace SearchAttributes fields only.
+	newRequest := *request
+	newRequest.SearchAttributes = mappedSearchAttributes
+	return &newRequest, nil
+}
+
+func (wh *WorkflowHandler) mapSignalWithStartWorkflowExecutionRequestSearchAttributes(request *workflowservice.SignalWithStartWorkflowExecutionRequest, namespaceName namespace.Name) (*workflowservice.SignalWithStartWorkflowExecutionRequest, error) {
+	mappedSearchAttributes, err := searchattribute.SubstituteAliases(wh.saMapper, request.GetSearchAttributes(), namespaceName.String())
+	if err != nil {
+		return nil, err
+	}
+	if mappedSearchAttributes == nil {
+		return request, nil
+	}
+
+	// Shallow copy request and replace SearchAttributes fields only.
+	newRequest := *request
+	newRequest.SearchAttributes = mappedSearchAttributes
+	return &newRequest, nil
+}
+
+func (wh *WorkflowHandler) mapCreateScheduleRequestSearchAttributes(request *workflowservice.CreateScheduleRequest, namespaceName namespace.Name) (*workflowservice.CreateScheduleRequest, error) {
+	mappedSearchAttributes, err := searchattribute.SubstituteAliases(wh.saMapper, request.GetSearchAttributes(), namespaceName.String())
+	if err != nil {
+		return nil, err
+	}
+	if mappedSearchAttributes == nil {
+		return request, nil
+	}
+
+	// Shallow copy request and replace SearchAttributes fields only.
+	newRequest := *request
+	newRequest.SearchAttributes = mappedSearchAttributes
+	return &newRequest, nil
+}
+
+func (wh *WorkflowHandler) mapCreateScheduleRequestStartWorkflowSearchAttributes(request *workflowservice.CreateScheduleRequest, namespaceName namespace.Name) (*workflowservice.CreateScheduleRequest, error) {
+	if startWorkflow := request.GetSchedule().GetAction().GetStartWorkflow(); startWorkflow != nil {
```

Comment:
flip this conditional so you can unindent most of this?

also this has to be in terms of schedulepb.Schedule, not CreateScheduleRequest, and one copy/replace has to move to the caller. you'll see when you merge...